NetCord/Gateway/GatewayClient.cs

```diff
@@ -17,71 +20,385 @@ public partial class GatewayClient : WebSocketClient, IEntity
+    /// </summary>
     public event Func<MessageDeleteEventArgs, ValueTask>? MessageDelete;
+
+    /// <summary>
+    /// Sent when multiple messages are deleted at once.
+    /// </summary>
     public event Func<MessageDeleteBulkEventArgs, ValueTask>? MessageDeleteBulk;
+
+    /// <summary>
+    /// Sent when a user adds a reaction to a message.
+    /// </summary>
     public event Func<MessageReactionAddEventArgs, ValueTask>? MessageReactionAdd;
+
+    /// <summary>
+    /// Sent when a user removes a reaction from a message.
+    /// </summary>
     public event Func<MessageReactionRemoveEventArgs, ValueTask>? MessageReactionRemove;
+
+    /// <summary>
+    /// Sent when a user explicitly removes all reactions from a message.
+    /// </summary>
     public event Func<MessageReactionRemoveAllEventArgs, ValueTask>? MessageReactionRemoveAll;
+
+    /// <summary>
+    /// Sent when a bot removes all instances of a given emoji from the reactions of a message.
+    /// </summary>
     public event Func<MessageReactionRemoveEmojiEventArgs, ValueTask>? MessageReactionRemoveEmoji;
+
+    /// <summary>
+    /// Sent when a user's presence or info, such as their name or avatar, is updated. Requires the <see cref="GatewayIntents.GuildPresences"/> intent.
+    /// The user object within this event can be partial, with the id being the only required field, everything else is optional. Along with this limitation, no fields are required, and the types of the fields are <b>not validated</b>. You should expect <b>any</b> combination of fields and types within this event.
+    /// </summary>
     public event Func<Presence, ValueTask>? PresenceUpdate;
+
+    /// <summary>
+    /// Sent when a user starts typing in a channel.
+    /// </summary>
     public event Func<TypingStartEventArgs, ValueTask>? TypingStart;
+
+    /// <summary>
+    /// Sent when properties about the current bot's user change.
+    /// Inner payload is a <see cref="CurrentUser"/> object.
+    /// </summary>
     public event Func<CurrentUser, ValueTask>? CurrentUserUpdate;
+
+    /// <summary>
+    /// Sent when someone joins/leaves/moves voice channels.
+    /// Inner payload is a <see cref="VoiceState"/> object.
+    /// </summary>
     public event Func<VoiceState, ValueTask>? VoiceStateUpdate;
+
+    /// <summary>
+    /// Sent when a guild's voice server is updated. This is sent when initially connecting to voice, and when the current voice instance fails over to a new server.
+    /// </summary>
     public event Func<VoiceServerUpdateEventArgs, ValueTask>? VoiceServerUpdate;
+
+    /// <summary>
+    /// Sent when a guild channel's webhook is created, updated, or deleted.
+    /// </summary>
     public event Func<WebhooksUpdateEventArgs, ValueTask>? WebhooksUpdate;
+
+    /// <summary>
+    /// Sent when a user uses an Application Command or Message Component.
```

Comment:
I don't think the description should be so specific. I would change it to
```suggestion
    /// Sent when a user uses an interaction.
```
I know that Discord documentation says "Application Command or Message Component", but I don't think it's correct because for example a modal submit interaction is neither of the options.

Reply:
Yeah that is a good point